### PR TITLE
Adds corpse spawners for every roundstart role

### PR DESCRIPTION
**Pull request by kilozombie**
image

image

Some roles were already ingame, but many were missing. The images above show every single new job as well as EVA-suited counterparts for some.

Admins and mappers can now spawn a corpse with appropriate clothing and ID from every roundstart role with substantially different clothing. This excludes some (but not all) alternate job titles if they don't have any real special features. In addition, corpses spawned in this way are now defaulted to having a random appearance. This is a global setting, and I decided it would be much more useful to have it on-- because while you can change it while making a new subtype, admins ingame can't change it for individually spawned corpses.

This means that admins can now spawn somebody of any role at roundstart, rejuv and do a few tweaks for realism, and blend in semi-seamlessly with the rest of the crew. This is usable with a lot of different possible bussing techniques and I'd be excited to see what admins do with it.

For mapping, this means making a derelict or vault with corpses from various departments is now completely possible. IDs will, however, link up with main station doors, so avoid using them too willy-nilly.

A subtype of the Assistant corpse, `assistant/robust`, spawns with various random contraband items and equipment that a very prolific member of the grey tide might have late into the round. They also automatically have all-access on their Assistant ID, so if bussing this in or putting it on a map, consider taking that away.

:cl:
 * rscadd: Admins and mappers can now spawn corpses of any roundstart role
 * tweak: Spawned corpses now have random appearance by default

### CODE REVIEW

File: code/modules/awaymissions/corpse.dm [hunk 1]
```diff
@@ -248,7 +248,44 @@
...
+	corpsesuit = list(/obj/item/clothing/suit/space/ancient, /obj/item/clothing/suit/space/rig/ror, /obj/item/clothing/suit/space/rig/syndi, /obj/item/clothing/suit/space/rig/atmos/, /obj/item/clothing/suit/armor/captain, /obj/item/clothing/suit/armor/vest, /obj/item/clothing/suit/armor/riot, /obj/item/clothing/suit/armor/laserproof, /obj/item/clothing/suit/armor/hos, /obj/item/clothing/suit/fire/firefighter, /obj/item/clothing/suit/space/rig/security, /obj/item/clothing/suit/space/rig/mining)
+	corpseshoes = list(/obj/item/clothing/shoes/magboots, /obj/item/clothing/shoes/magboots/atmos, /obj/item/clothing/shoes/magboots/captain, /obj/item/clothing/shoes/jackboots/knifeholster, /obj/item/clothing/shoes/galoshes)
+	corpsegloves = list(/obj/item/clothing/gloves/yellow/power, /obj/item/clothing/gloves/yellow/power, /obj/item/clothing/gloves/yellow/power, /obj/item/clothing/gloves/yellow/power, /obj/item/clothing/gloves/black, /obj/item/clothing/gloves/captain)
```
Comment (I-VAPE-VOX-CLOACA-EVERY-DAY-OF-MY-LIFE) [inline code/modules/awaymissions/corpse.dm]: Yeah these aren't insulateds these are the emperor palpatine tator engineer gloves

Comment (kilozombie) [reply code/modules/awaymissions/corpse.dm]: Oh shit

File: code/modules/awaymissions/corpse.dm [hunk 2]
```diff
@@ -262,6 +299,146 @@
...
+	corpseidjob = "Bridge Officer"
+
+/obj/effect/landmark/corpse/chaplain
+	name = "Chaplain"
+	corpseuniform = /obj/item/clothing/under/rank/chaplain
+	corpseback = /obj/item/weapon/storage/backpack
+	corpseradio = /obj/item/device/radio/headset
+	corpseshoes = /obj/item/clothing/shoes/laceup
+	corpseid = 1
+	corpseidjob = "Captain"
```
Comment (Durgen) [inline code/modules/awaymissions/corpse.dm]: is the chaplain meant to have a cap titled ID

Comment (kilozombie) [reply code/modules/awaymissions/corpse.dm]: definitely not

**Comment by small-boss** [comment]
do these corpses also get random names?

**Comment by kilozombie** [comment]
Not by default, they'll be named after their job. You can change this in a subtype but I figured that since admins can actually rename stuff on the fly, it's not a big deal.

File: code/modules/awaymissions/corpse.dm [hunk 3]
```diff
@@ -248,7 +248,44 @@
...
+	corpseuniform = /obj/item/clothing/under/color/grey
+	corpseshoes = /obj/item/clothing/shoes/black
+	corpsebelt = /obj/item/weapon/storage/bag/plasticbag
+	corpseback = /obj/item/weapon/storage/backpack
+	corpseradio = /obj/item/device/radio/headset
+	corpseid = 1
+	corpseidjob = "Assistant"
+	corpseidaccess = "Assistant"
+
+/obj/effect/landmark/corpse/assistant/robust
```
Comment (jknpj) [inline code/modules/awaymissions/corpse.dm:264]: Jesus Christ. Monthly remember that we need outfit datums to fix this kind of shit.

Comment (kilozombie) [reply code/modules/awaymissions/corpse.dm:264]: What's wrong with it besides being annoyingly large

Comment (jknpj) [reply code/modules/awaymissions/corpse.dm:264]: Our job code has its own outfit clusterfuck
Antag Madness has one of their own
Equip Loadout verb also has one
ERT/Squid/Elites ditto
Landmarks, too

See the issue? But that's not your fault.

Comment (kilozombie) [reply code/modules/awaymissions/corpse.dm:264]: That makes sense. Having it be datumized? would help with a LOT of things.

I was looking up object paths to get these corpses equipped using all of the job code files and it definitely is a mess. I was actually under the impression it'd be more centralized than "okay, literally just slap this equipment on their body or something"

**Pull request merged**